db/db_compaction_test.cc

```diff
@@ -1371,6 +1412,68 @@ TEST_F(DBCompactionTest, RecoverDuringMemtableCompaction) {
   } while (ChangeOptions());
 }
 
+TEST_F(DBCompactionTest, CompactionWithDeletionsAndMinFileSize) {
+  Options options = CurrentOptions();
+  options.compaction_style = kCompactionStyleLevel;
+  options.write_buffer_size = 1024 * 1024;  // 1MB
+  options.level0_file_num_compaction_trigger = 100;
+
+  const uint64_t kMinFileSize = 32 * 1024;  // 32KB
+  options.table_properties_collector_factories = {
+      NewCompactOnDeletionCollectorFactory(100, 50, 0.5, kMinFileSize)};
+  auto listener = new DeletionTriggeredCompactionWithMinFileSizeTestListener();
+  options.listeners.emplace_back(listener);
+
+  DestroyAndReopen(options);
+
+  // Create a large file with puts
+  Random rnd(301);
+  for (int i = 0; i < 100; i++) {
+    ASSERT_OK(Put(Key(i), rnd.RandomString(1024)));
+  }
+  ASSERT_OK(Flush());
+
+  std::vector<LiveFileMetaData> initial_metadata;
+  db_->GetLiveFilesMetaData(&initial_metadata);
+  ASSERT_GT(initial_metadata.size(), 0);
+
+  // Create a small file less than kMinFileSize with deletions that shouldn't
+  // trigger compaction
```

Comment:
Is it intentional to mix the testing of the large file case and the small file case at once (i.e, in one db session and experience the compaction picking the same time)? If not, can we make the test case more straightforward by separating these two cases with DestroyAndReopen()?

Reply:
I see - replied in below https://github.com/facebook/rocksdb/pull/13825#discussion_r2252515076